dgp/core/controllers/project_controllers.py

```diff
@@ -0,0 +1,292 @@
+
+    def set_name(self):  # pragma: no cover
+        new_name = get_input("Set Project Name", "Enter a Project Name", self.project.name)
+        if new_name:
+            self.set_attr('name', new_name)
+
+    def show_in_explorer(self):  # pragma: no cover
+        # TODO Linux KDE/Gnome file browser launch
+        ppath = str(self.project.path.resolve())
+        if sys.platform == 'darwin':
+            script = 'oascript'
+            args = '-e tell application \"Finder\" -e activate -e select POSIX file \"' + ppath + '\" -e end tell'
+        elif sys.platform == 'win32':
+            script = 'explorer'
+            args = shlex.quote(ppath)
+        else:
+            self.log.warning("Platform %s is not supported for this action.", sys.platform)
+            return
+
+        QProcess.startDetached(script, shlex.split(args))
+
+    # TODO: What to do about these dialog methods - it feels wrong here
+    def add_flight(self):  # pragma: no cover
+        dlg = AddFlightDialog(project=self, parent=self.get_parent_widget())
+        return dlg.exec_()
+
+    def add_gravimeter(self):  # pragma: no cover
+        """Launch a Dialog to import a Gravimeter configuration"""
+        dlg = AddGravimeterDialog(self, parent=self.get_parent_widget())
+        return dlg.exec_()
+
+    def update(self):  # pragma: no cover
+        """Emit an update event from the parent Model, signalling that
+        data has been added/removed/modified in the project."""
+        self.setText(self._project.name)
+        if self.model() is not None:
+            self.model().project_changed.emit()
+
+    def _post_load(self, datafile: DataFile, dataset: IDataSetController,
+                   data: DataFrame) -> None:  # pragma: no cover
+        """
+        This is a slot called upon successful loading of a DataFile by a
+        FileLoader Thread.
+
+        Parameters
+        ----------
+        datafile : :obj:`dgp.core.models.data.DataFile`
+            The DataFile reference object to be processed
+        data : DataFrame
+            The ingested pandas DataFrame to be dumped to the HDF5 store
+
+        """
+        # TODO: Insert DataFile into appropriate child
+        datafile.set_parent(dataset)
+        if HDF5Manager.save_data(data, datafile, path=self.hdf5path):
+            self.log.info("Data imported and saved to HDF5 Store")
+        dataset.add_datafile(datafile)
+        return
+
+    def load_file_dlg(self, datatype: DataTypes = DataTypes.GRAVITY,
+                      flight: IFlightController = None,
+                      dataset: IDataSetController = None) -> None:  # pragma: no cover
+        """
+        Project level dialog for importing/loading Gravity or Trajectory data
+        files. The Dialog generates a DataFile and a parameter map (dict) which
+        is passed along to a FileLoader thread to ingest the raw data-file.
+        On completion of the FileLoader, AirborneProjectController._post_load is
```

Comment:
Does this still handle aligning the trajectory and gravity dataframes after they have both been imported?

Reply:
It doesn't, I don't think I re-implemented the aligning functionality yet as I wasn't sure of the best place to do it.
I think the most logical place to do the align is within the DataSet Controller, probably called from the add_datafile method (see dataset_controller.py line 160).

Let me know your thoughts on that; it can be re-implemented pretty easily from the old code so I can add the change in this PR.

Reply:
I agree with the placement of that functionality, and I saw your comments in the code later on. You can leave it out of this PR and we can put it back in as we start to piece the processing workflow together.